bin/arnold

```diff
@@ -320,27 +335,71 @@ function _set_k8s_env() {
         source "${customer_vars}"
     fi
 
-    # Check that the current user is already logged in k8s. Note that the
-    # `whoami` command is only compatible with openshift and should be adapted
-    # for k8s.
-    if ! ${KUBECTL} whoami &> /dev/null ; then
-        if [[ -z "${K8S_SA_TOKEN}" ]] ; then
+    # If an authentication token is provided, check that the K8S_AUTH_HOST is also provided.
+    if [[ -n "${K8S_AUTH_API_KEY}" && -z "${K8S_AUTH_HOST}" ]] ; then
+      log error "Error: you should provide K8S_AUTH_API_KEY and K8S_AUTH_HOST"
+      exit 1
+    fi
+
+    # If no authentication token is provided, we'll get it from the K8S_SERVICE_ACCOUNT
+    # service account in the configuration context K8S_CONTEXT provided.
+    if [[ -z "${K8S_AUTH_API_KEY}" ]] ; then
+      if [[ -z "${K8S_CONTEXT}" ]]; then
+        log error "Error: you should provide a K8S_CONTEXT or K8S_AUTH_API_KEY + K8S_AUTH_HOST."
+        exit 1
+      fi
+
+      # Ensure that the context K8S_CONTEXT exist
+      if ! ${KUBECTL} config get-contexts -o=name | grep -q "^${K8S_CONTEXT}\$" ; then
+        log error "Error: unable to find context ${K8S_CONTEXT} in kubernetes configuration."
+        exit 1
+      fi
+
+      # Get K8S_AUTH_HOST from configuration
+      K8S_AUTH_HOST=$(${KUBECTL} --namespace "${K8S_SERVICE_ACCOUNT_NAMESPACE}" config view -o jsonpath="{.clusters[?(@.name==\"$K8S_CONTEXT\")].cluster.server}")
+      # Fix K8S_AUTH_HOST if it contains a loopback IP address
+      K8S_AUTH_HOST=$(echo "${K8S_AUTH_HOST}" | sed -r 's/localhost|0(\.0){3}|127(\.[0-9]+){3}/'"${K8S_DOMAIN}"'/')
```

Comment:
can you explain more in the comment?

Reply:
Done, please tell me if it's still not clear.